client/post-editor/media-modal/index.jsx

```diff
@@ -213,13 +214,10 @@ export class EditorMediaModal extends Component {
 			return;
 		}
 
-		let toDelete = selectedItems;
-		if ( ModalViews.DETAIL === this.props.view ) {
-			toDelete = toDelete[ this.getDetailSelectedIndex() ];
-			this.setNextAvailableDetailView();
-		}
-
-		this.props.deleteMedia( site.ID, toDelete );
+		this.props.deleteMedia(
```

Comment:
Please note I removed the detail view check (and related tests), because apparently it's not possible to delete an item from the detail view when using the media modal (not the media library, there it's possible). It seems this functionality was removed quite a while ago.